protocol/messenger_activity_center.go

```diff
@@ -60,16 +63,70 @@ func (m *Messenger) ActivityCenterNotificationsCount(request ActivityCenterCount
 }
 
 func (m *Messenger) HasUnseenActivityCenterNotifications() (bool, error) {
-	return m.persistence.HasUnseenActivityCenterNotifications()
+	seen, _, err := m.persistence.HasUnseenActivityCenterNotifications()
+	return seen, err
 }
 
 func (m *Messenger) GetActivityCenterState() (*ActivityCenterState, error) {
 	return m.persistence.GetActivityCenterState()
 }
 
+func (m *Messenger) syncActivityCenterNotificationState(state *ActivityCenterState) error {
+	if state == nil {
+		return nil
+	}
+	syncStateMessage := &protobuf.SyncActivityCenterNotificationState{
+		UpdatedAt: state.UpdatedAt,
+		HasSeen:   state.HasSeen,
+	}
+	encodedMessage, err := proto.Marshal(syncStateMessage)
+	if err != nil {
+		return err
+	}
+	return m.sendToPairedDevices(context.TODO(), common.RawMessage{
+		Payload:             encodedMessage,
+		MessageType:         protobuf.ApplicationMetadataMessage_SYNC_ACTIVITY_CENTER_NOTIFICATION_STATE,
+		ResendAutomatically: true,
+	})
+}
+
+func (m *Messenger) syncActivityCenterNotifications(notifications []*ActivityCenterNotification) (err error) {
+	if notifications == nil {
+		return nil
+	}
+	var s []*protobuf.SyncActivityCenterNotification
+	for _, n := range notifications {
+		if n == nil {
```

Comment:
I checked all calls to `syncActivityCenterNotifications`, and in all cases the slice can't have nils inside. Maybe I missed something. Could you please share the offending call that can pass a slice of notifications with nil(s) inside?

Unexpected nils can be a more serious problem in the system, so if we can pinpoint the place that's building the `notifications` slice with nils, I think we should fix it, and not allow the slice to propagate further in the system.

Reply:
my bad, i'll remove this as it confused you 😅

Reply:
It's not as if it's causing a big problem @qfrank, but more on the side of we usually don't need such checks when iterating because most slices are made of non-nil elements. Thanks for taking this in consideration.

Reply:
Thank you for your valuable review👍